geomstats/geometry/pullback_metric.py

```diff
@@ -86,81 +51,48 @@ def metric_matrix(self, base_point=None, n_jobs=1, **joblib_kwargs):
         mat : array-like, shape=[..., dim, dim]
             Inner-product matrix.
         """
-        immersed_base_point = self.immersion(base_point)
-        jacobian_immersion = self.jacobian_immersion(base_point)
-        basis_elements = gs.eye(self.dim)
-
-        @joblib.delayed
-        @joblib.wrap_non_picklable_objects
-        def pickable_inner_product(i, j):
-            immersed_basis_element_i = gs.squeeze(
-                gs.matvec(jacobian_immersion, basis_elements[i])
-            )
-            immersed_basis_element_j = gs.squeeze(
-                gs.matvec(jacobian_immersion, basis_elements[j])
-            )
-            return self.embedding_metric.inner_product(
-                immersed_basis_element_i,
-                immersed_basis_element_j,
-                base_point=immersed_base_point,
-            )
+        dim, dim_embedding = self._space.dim, self._space.embedding_space.dim
+        is_vec = check_is_batch(self._space, base_point)
 
-        pool = joblib.Parallel(n_jobs=n_jobs, **joblib_kwargs)
-        out = pool(
-            pickable_inner_product(i, j)
-            for i, j in itertools.product(range(self.dim), range(self.dim))
-        )
-        metric_mat = gs.reshape(gs.array(out), (-1, self.dim, self.dim))
-        return metric_mat[0] if base_point.ndim == 1 else metric_mat
+        immersed_base_point = self._space.immersion(base_point)
```

Comment:
it is thus assumed that the space has an immersion function. i.e. is an immersed set. LGTM, I just wonder if we could make it explicit in the documentation somewhere that this is how it works?

Reply:
There's a notebook that shows how to define this. It is actually expected this to be an `ImmersedSet`. (see below)